hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/ComplexKeyGenerator.java

```diff
@@ -37,14 +36,10 @@ public class ComplexKeyGenerator extends BuiltinKeyGenerator {
 
   public ComplexKeyGenerator(TypedProperties props) {
     super(props);
-    this.recordKeyFields = Arrays.stream(props.getString(KeyGeneratorOptions.RECORDKEY_FIELD_NAME.key()).split(","))
-        .map(String::trim)
-        .filter(s -> !s.isEmpty())
-        .collect(Collectors.toList());
-    this.partitionPathFields = Arrays.stream(props.getString(KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME.key()).split(","))
-        .map(String::trim)
-        .filter(s -> !s.isEmpty())
-        .collect(Collectors.toList());
+    this.recordKeyFields = props.getStringList(KeyGeneratorOptions.RECORDKEY_FIELD_NAME.key(), ",",
+        Collections.singletonList(KeyGeneratorOptions.RECORDKEY_FIELD_NAME.defaultValue()));
```

Comment:
This behavior change may not be compatible. `RECORDKEY_FIELD_NAME` has a default value. In the unlikely event that user does not set this config, it will result in `uuid` being default and this field may not be part of schema. Without this change, an exception would be thrown.

Reply:
hi @codope, I think this change is backwards compatible. Given a config that works in <0.11.0, the behaviour is the same after this change.

Assume that `uuid` is not part of the Schema, and the user did not set `RECORDKEY_FIELD_NAME`.

- Old behaviour: keygen throws an exception due to unset config
- New behaviour: the write fails due to an invalid record key

Wouldn't the new failure mode make more sense considering that `RECORDKEY_FIELD_NAME` actually has a default value?

In any case, I can re-scope this change to only fix hive-sync if needed.

Reply:
It would be better to not rely on default value for record key. See HUDI-3456 for example.
> In any case, I can re-scope this change to only fix hive-sync if needed.

That would be much preferable.

Reply:
in that case, already fixed by #5585, will close this